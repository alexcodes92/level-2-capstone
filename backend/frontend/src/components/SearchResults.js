import React from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

import SearchResultCard from "./SearchResultCard";

const SearchResults = ({ results, favourites, toggleFavourite }) => {
  // loops through the search results and renders card for each
  if (results) {
    return (
      <Container className="mb-4 mt-4">
        <Row>
          {results.map((result) => {
            const isFavourite = favourites.find(
              (favourite) => favourite.id === result.trackId
            );
            return (
              <Col mb={2} xs={6} lg={4} key={result.trackId}>
                <SearchResultCard
                  id={result.trackId}
                  artist={result.artistName}
                  track={result.trackCensoredName}
                  image={result.artworkUrl100}
                  url={result.trackViewUrl}
                  isFavourited={isFavourite !== undefined}
                  toggleFavourite={toggleFavourite}
                />
              </Col>
            );
          })}
        </Row>
      </Container>
    );
  }

  return null;
};

export default SearchResults;
