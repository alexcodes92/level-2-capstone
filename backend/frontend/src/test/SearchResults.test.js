import React from "react";
import SearchResults from "../components/SearchResults";
import renderer from "react-test-renderer";

const results = [
  {
    trackId: 2,
    artistName: "Foo Fighters",
    trackCensoredName: "Everlong",
    artworkUrl100: "image-url",
    trackViewUrl: "url",
  },
];

test("snapshot test", () => {
  const tree = renderer
    .create(<SearchResults results={results} favourites={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
