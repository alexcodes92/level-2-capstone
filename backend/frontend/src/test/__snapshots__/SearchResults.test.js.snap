// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="mb-4 mt-4 container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4 col-6"
      mb={2}
    >
      <div
        className="mb-3 card"
      >
        <img
          className="card-img-top"
          src="image-url"
        />
        <div
          className="card-body"
        >
          <a
            className="card-link"
            href="url"
            target="_blank"
          >
            <div
              className="card-title h5"
            >
              Everlong
            </div>
          </a>
          <div
            className="mb-4 card-subtitle h6"
          >
            Foo Fighters
          </div>
          <button
            className="p-2 btn btn-light"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add to favourites
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
